### ISSUE TITLE
Fix `RuleContext['fix']` type

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#8308]
### Details

https://github.com/stylelint/stylelint/blob/8a173d4ad4164aae9160b69b248de573bd2235d9/types/stylelint/index.d.ts#L234

does not reflect

https://github.com/stylelint/stylelint/blob/8a173d4ad4164aae9160b69b248de573bd2235d9/lib/lintPostcssResult.mjs#L92

https://github.com/stylelint/stylelint/blob/8a173d4ad4164aae9160b69b248de573bd2235d9/lib/lintPostcssResult.mjs#L106

see also https://github.com/stylelint/stylelint/pull/8106#discussion_r1845183846

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.13.0"
}
```

### What did you expect to happen?

no semantic error TS2322

### What actually happened?

It was detected the 20/12/2024 by [this job](https://github.com/stylelint/stylelint-ecosystem-tester/actions/runs/12423419685/job/34686910033) but we overlooked it.
i.e. `stylelint-ecosystem-tester` did its job

```
 semantic error TS2322: Type '{ (properties: PrimaryOptions, options: SecondaryOptions, context?: RuleContext): (root: Root, result: PostcssResult) => void; primaryOptionArray: boolean | undefined; ruleName: string; messages: stylelint.RuleMessages; meta: stylelint.RuleMeta | undefined; }' is not assignable to type 'StylelintPlugin<PrimaryOptions, SecondaryOptions>'.
  Type '{ (properties: PrimaryOptions, options: SecondaryOptions, context?: RuleContext): (root: Root, result: PostcssResult) => void; primaryOptionArray: boolean | undefined; ruleName: string; messages: stylelint.RuleMessages; meta: stylelint.RuleMeta | undefined; }' is not assignable to type 'RuleBase<PrimaryOptions, SecondaryOptions>'.
    Types of parameters 'context' and 'context' are incompatible.
      Type 'import("/…/index").RuleContext' is not assignable to type 'RuleContext'.
        Types of property 'fix' are incompatible.
          Type 'boolean | FixMode | undefined' is not assignable to type 'boolean | undefined'.
            Type '"lax"' is not assignable to type 'boolean | undefined'.
```

### Do you have a proposal to fix the bug?

1. [x] revert to `fix?: boolean | undefined;`
probably introduced by [this](https://github.com/stylelint/stylelint/pull/8106/files#diff-9a4b261e295bef1ccce813aa194ac4023a1dcf98236667889964b1ba6fbc551fR232)
2. [x] create an issue for `lax`/`strict` `configFile`/`config` support
    - #8333
